Sources/SwiftInspectorCore/PropertyAnalyzer.swift

```diff
@@ -0,0 +1,248 @@
+// Created by Tyler Hedrick on 8/14/20.
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import SwiftSyntax
+
+// MARK: - TypeLocationAnalyzer
+
+public final class PropertyAnalyzer: Analyzer {
+
+  /// - Parameter typeName: The name of the type to get property information for
+  /// - Parameter cachedSyntaxTree: The cached syntax tree to return the AST tree from
+  public init(typeName: String, cachedSyntaxTree: CachedSyntaxTree = .init()) {
+    self.typeName = typeName
+    self.cachedSyntaxTree = cachedSyntaxTree
+  }
+
+  /// Finds the location(s) of the specified type name in the Swift file
+  /// - Parameter fileURL: The fileURL where the Swift file is located
+  public func analyze(fileURL: URL) throws -> TypeWithPropInfo? {
+    let syntax: SourceFileSyntax = try cachedSyntaxTree.syntaxTree(for: fileURL)
+    var result: TypeWithPropInfo?
+    let visitor = TypeSyntaxVisitor() { [unowned self] typeWithPropInfo in
+      guard self.typeName == typeWithPropInfo.name else { return }
+      result = try? typeWithPropInfo.merge(with: result)
+    }
+    visitor.walk(syntax)
+    return result
+  }
+
+  // MARK: Private
+
+  private let cachedSyntaxTree: CachedSyntaxTree
+  private let typeName: String
+}
+
+// MARK: - TypeLocationSyntaxVisitor
+
+private final class TypeSyntaxVisitor: SyntaxVisitor {
+
+  init(onNodeVisit: @escaping (_ info: TypeWithPropInfo) -> Void) {
+    self.onNodeVisit = onNodeVisit
+  }
+
+  override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
```

Comment:
Should we validate in here the name instead that after doing all the processing? That will likely improve the performance of this tool (since we won't be processing every type, only the types we care about)

I'mm thinking we should pass the type name to the `TypeSyntaxVisitor` and if the name is not the one expected on these methods we can do a guard and visit children immediately, and only process the node when the type name is the one we care about.

```swift
guard node.identifier.text == typeName else { return .visitChildren }
```